kpatch-build/create-diff-object.c

```diff
@@ -608,15 +637,41 @@ static void kpatch_compare_sections(struct list_head *seclist)
 	/* sync symbol status */
 	list_for_each_entry(sec, seclist, list) {
 		if (is_rela_section(sec)) {
-			if (sec->base->sym && sec->base->sym->status != CHANGED)
+			if (sec->base->sym && sec->base->sym->status != CHANGED) {
 				sec->base->sym->status = sec->status;
+				if (sec->base->sym->parent)
+					sec->base->sym->parent->status = sec->status;
+			}
 		} else {
-			if (sec->sym && sec->sym->status != CHANGED)
+			if (sec->sym && sec->sym->status != CHANGED) {
 				sec->sym->status = sec->status;
+				if (sec->sym->parent)
+					sec->sym->parent->status = sec->status;
+			}
 		}
 	}
 }
 
+static sstype_t kpatch_section_type(struct section *sec)
+{
+	if (strncmp(sec->name, ".text.", 6)) {
+		DIFF_FATAL("unsupported section type: %s", sec->name);
+	}
```

Comment:
I think this DIFF_FATAL could be confusing, based on how this function is called.  For example, if were a data section, this error would hit instead of the "symbol changed sections" error.  So maybe just remove this check altogether and assume that it's just always SUBSECTION_NORMAL if it doesn't match the other string comparisons?